src/controller/TypedReadCallback.h

```diff
@@ -153,6 +153,19 @@ class TypedReadEventCallback final : public app::ReadClient::Callback
 
     void OnError(const app::ReadClient * apReadClient, CHIP_ERROR aError) override { mOnError(nullptr, aError); }
 
+    void OnDeallocatePaths(chip::app::ReadPrepareParams && aReadPrepareParams) override
+    {
+        if (aReadPrepareParams.mpAttributePathParamsList != nullptr)
+        {
+            delete[] aReadPrepareParams.mpAttributePathParamsList;
```

Comment:
This just doesn't feel very nice/right to me. I think it's best if we just permitted ReadClient to call Platform::New for these attribute path params...

Reply:
Fully agree on it, thanks to point out it. Just Use platform::New and platform::Delete on it, thanks